AmplifyPlugins/DataStore/AWSDataStoreCategoryPluginAuthIntegrationTests/AWSDataStoreCategoryPluginAuthIntegrationTests.swift

```diff
@@ -0,0 +1,230 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+import AmplifyPlugins
+import AWSDataStoreCategoryPlugin
+
+@testable import Amplify
+@testable import AmplifyTestCommon
+
+class AWSDataStoreCategoryPluginAuthIntegrationTests: XCTestCase {
+
+    struct User {
+        let username: String
+        let password: String
+    }
+
+    let amplifyConfigurationFile = "AWSDataStoreCategoryPluginAuthIntegrationTests-amplifyconfiguration"
+    let credentialsFile = "AWSDataStoreCategoryPluginAuthIntegrationTests-credentials"
+    var user1: User!
+    var user2: User!
+
+    let syncReceived = HubPayload.EventName.DataStore.syncReceived
+    let syncStarted = HubPayload.EventName.DataStore.syncStarted
+
+    override func setUp() {
+        Amplify.Logging.logLevel = .verbose
+
+        do {
+            let credentials = try TestConfigHelper.retrieveCredentials(forResource: credentialsFile)
+
+            guard let user1 = credentials["user1"],
+                let user2 = credentials["user2"],
+                let passwordUser1 = credentials["passwordUser1"],
+                let passwordUser2 = credentials["passwordUser2"] else {
+                    XCTFail("Missing credentials.json data")
+                    return
+            }
+
+            self.user1 = User(username: user1, password: passwordUser1)
+            self.user2 = User(username: user2, password: passwordUser2)
+
+            try Amplify.add(plugin: AWSDataStorePlugin(modelRegistration: SocialNoteModelRegistration()))
+            try Amplify.add(plugin: AWSAPIPlugin())
+            try Amplify.add(plugin: AWSCognitoAuthPlugin())
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(forResource: amplifyConfigurationFile)
+            try Amplify.configure(amplifyConfig)
+
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+        if isSignedIn() {
+            signOut()
+        }
```

Comment:
is there any reason for this... because we are calling tearDown() which has signOut and reset?

Reply:
this was to prevent previous test executions failures so tearDown isn't called, or tearDown() is called but failed for some reason. i think this is applicable for a single test run as well since one test method may fail while the next one continues

Reply:
hmmm.. if teardown isnt' being called on a single test run, then should we just add signout to each one of the integration tests?

Reply:
teardown should be called on a single test method run. i was referring to more of a scenario where test is stopped, say i manually terminate it by pressing Stop, then tearDown is never run